Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -366,15 +366,15 @@ extension HTTPClient {
     /// `EventLoopFuture<Response>` of the execution or cancellation of the execution.
     public final class Task<Response> {
         /// `EventLoop` used to execute and process this request.
-        public let eventLoop: EventLoop
+        public internal(set) var currentEventLoop: EventLoop
         let promise: EventLoopPromise<Response>
 
         private var channel: Channel?
         private var cancelled: Bool
         private let lock: Lock
 
-        public init(eventLoop: EventLoop) {
-            self.eventLoop = eventLoop
+        init(eventLoop: EventLoop) {
```

Comment:
Just like changing `eventLoop` to `currentEventLoop`, I can also make this change (from `public` to `internal`) in another PR if preferred although I thought it might be acceptable here since it's closely related to the issue this PR fixes.